### PR TITLE
Http adapter: Check device registration during command response upload

### PR DESCRIPTION
**Pull request by kaniyan**

### CODE REVIEW

File: adapters/http-vertx-base/src/test/java/org/eclipse/hono/adapter/http/AbstractVertxBasedHttpProtocolAdapterTest.java [hunk 1]
```diff
@@ -410,6 +411,42 @@ public void testUploadCommandResponseFailsForDisabledTenant() {
...
+                .put(TenantConstants.FIELD_ADAPTERS_TYPE, ADAPTER_TYPE)
+                .put(TenantConstants.FIELD_ENABLED, true));
+        when(tenantClient.get(eq("tenant"), (SpanContext) any())).thenReturn(Future.succeededFuture(to));
+
+        // which is connected to a Credentials service that has credentials on record for device 9999
+        when(adapter.getAuthenticatedDevice(ctx)).thenReturn(new DeviceUser("tenant", "9999"));
+
+        // but for which no registration information is available
+        when(regClient.assertRegistration((String) any(), (String) any(), (SpanContext) any()))
+                .thenReturn(Future.failedFuture(new ClientErrorException(HttpURLConnection.HTTP_FORBIDDEN,
```
Comment (sophokles73) [inline adapters/http-vertx-base/src/test/java/org/eclipse/hono/adapter/http/AbstractVertxBasedHttpProtocolAdapterTest.java]: this should probably return a 404, reflecting the case that the device does not (or no longer) exist anymore, right?

File: adapters/http-vertx-base/src/test/java/org/eclipse/hono/adapter/http/AbstractVertxBasedHttpProtocolAdapterTest.java [hunk 2]
```diff
@@ -410,6 +411,42 @@ public void testUploadCommandResponseFailsForDisabledTenant() {
         verify(metrics, never()).incrementCommandResponseDeliveredToApplication(eq("tenant"));
     }
 
+    /**
+     * Verifies that the adapter fails the upload of a command response with a 403 if the device is not registered and
+     * the valid credentials for the device is available.
+     */
+    @Test
+    public void testUploadCommandResponseFailsForDeletedDevices() {
```
Comment (sophokles73) [inline adapters/http-vertx-base/src/test/java/org/eclipse/hono/adapter/http/AbstractVertxBasedHttpProtocolAdapterTest.java]: the generic case that the device does not exist, isn't it?
So, I would propose to name this method *testUploadCommandResponseFailsForNonExistingDevice*

File: adapters/http-vertx-base/src/test/java/org/eclipse/hono/adapter/http/AbstractVertxBasedHttpProtocolAdapterTest.java (hunk 1 above)
Comment (kaniyan) [reply adapters/http-vertx-base/src/test/java/org/eclipse/hono/adapter/http/AbstractVertxBasedHttpProtocolAdapterTest.java]: Device registration doesn't exist and where as device credentials exists. It seemed to me partially available and hence it led me to 403. I think so too, that 404 is more appropriate.

File: adapters/http-vertx-base/src/test/java/org/eclipse/hono/adapter/http/AbstractVertxBasedHttpProtocolAdapterTest.java (hunk 2 above)
Comment (kaniyan) [reply adapters/http-vertx-base/src/test/java/org/eclipse/hono/adapter/http/AbstractVertxBasedHttpProtocolAdapterTest.java]: Done

File: adapters/http-vertx-base/src/test/java/org/eclipse/hono/adapter/http/AbstractVertxBasedHttpProtocolAdapterTest.java [hunk 3]
```diff
@@ -410,6 +411,42 @@ public void testUploadCommandResponseFailsForDisabledTenant() {
         verify(metrics, never()).incrementCommandResponseDeliveredToApplication(eq("tenant"));
     }
 
+    /**
+     * Verifies that the adapter fails the upload of a command response with a 403 if the device is not registered and
```
Comment (sophokles73) [inline adapters/http-vertx-base/src/test/java/org/eclipse/hono/adapter/http/AbstractVertxBasedHttpProtocolAdapterTest.java]: now the JavaDoc needs to be adapted to the new status code as well ...

Comment (kaniyan) [reply adapters/http-vertx-base/src/test/java/org/eclipse/hono/adapter/http/AbstractVertxBasedHttpProtocolAdapterTest.java]: Done

**Pull request merged**